frontend-react/src/shared/Tooltip.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { describe, test } from "@jest/globals";
+import { screen } from "@testing-library/react";
+import React from "react";
+import user from "@testing-library/user-event";
+
+import { renderApp } from "../utils/CustomRenderUtils";
+
+import {
+    Tooltip,
+    useTooltipContext,
+    TooltipContext,
+    TooltipContextType,
+    TooltipTrigger,
+    TooltipContent,
+} from "./Tooltip";
+
+const TooltipContextReceiverComponent = () => {
+    const { placement } = useTooltipContext();
+    return <>{placement}</>;
+};
+
+const TooltipContextCreatorComponent = ({
+    children,
+}: {
+    children: React.ReactNode;
+}) => {
+    const ctx: TooltipContextType = {
+        placement: "Test",
+    } as any;
+    return (
+        <TooltipContext.Provider value={ctx}>
+            {children}
+        </TooltipContext.Provider>
+    );
+};
+
+describe("Tooltip", () => {
+    describe("useTooltipContext", () => {
+        test("no context - error", () => {
+            expect(() =>
+                renderApp(<TooltipContextReceiverComponent />)
+            ).toThrowError("Tooltip components must be wrapped in <Tooltip />");
+        });
+
+        test("context", () => {
+            renderApp(
+                <TooltipContextCreatorComponent>
+                    <TooltipContextReceiverComponent />
+                </TooltipContextCreatorComponent>
+            );
+            expect(screen.getByText("Test")).toBeInTheDocument();
+        });
+    });
+
+    describe("Tooltip", () => {
+        test("creates tooltip context", () => {
+            renderApp(
+                <Tooltip placement={"Test" as any}>
```

Comment:
Should this be allowed or should it be limited to the four possible options?

Reply:
I should've just followed the typing and passed a valid string like bottom but it's obvious I was brain fried at this point haha. Will fix.

Reply:
No problem!  I definitely understand 😆